Shopify/ruby-lsp
# Add `watched_files_dynamic_registration` to global state

### Motivation
Work towards Shopify/ruby-lsp-rails#326. `ruby-lsp-rails` wants to watch for file changes to `**/structure.sql`. This allows it to check if the client supports this functionality before trying to send the request.

It makes sense to add docs on how/why you might make use of this option in an addon, but I'll defer that until after I've actually written some functioning code.

### Implementation

Move the options extraction for this into the global state.

### Automated Tests

Add a few variants that a client might send.

### Manual Tests

Edit a ruby file and observe that the file watcher is still being registered, i.e. that the index is being updated.
